kvserver: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic is flaky

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/17413258?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true


```
=== RUN   TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic
    datadriven.go:144:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3754/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/testdata/flow_control_integration_v2/raft_membership_v2_encoding_apply_to_elastic:1:
        output didn't match expected:
        @@ -59,14 +59,10 @@
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
             FROM crdb_internal.kv_flow_control_handles_v2
           range_id | store_id | total_tracked_tokens
         -----------+----------+-----------------------
        -  70       | 1        | 2.0 MiB
        -  70       | 2        | 2.0 MiB
        -  70       | 3        | 2.0 MiB
        -  70       | 4        | 1.0 MiB
         -- (Removing voting replica from n3.)
        @@ -83,13 +79,13 @@
         SELECT range_id, store_id, crdb_internal.humanize_bytes(total_tracked_tokens::INT8)
             FROM crdb_internal.kv_flow_control_handles_v2
           range_id | store_id | total_tracked_tokens
         -----------+----------+-----------------------
        -  70       | 1        | 3.0 MiB
        -  70       | 2        | 3.0 MiB
        -  70       | 4        | 2.0 MiB
        +  70       | 1        | 1.0 MiB
        +  70       | 2        | 1.0 MiB
        +  70       | 4        | 1.0 MiB
           70       | 5        | 1.0 MiB
         -- (Allow below-raft admission to proceed.)
        @@ -115,11 +111,11 @@
         ORDER BY name ASC;
           kvflowcontrol.tokens.eval.elastic.available                       | 40 MiB
           kvflowcontrol.tokens.eval.elastic.deducted                        | 11 MiB
           kvflowcontrol.tokens.eval.elastic.returned                        | 11 MiB
        -  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 2.0 MiB
        +  kvflowcontrol.tokens.eval.elastic.returned.disconnect             | 7.0 MiB
           kvflowcontrol.tokens.eval.elastic.unaccounted                     | 0 B
           kvflowcontrol.tokens.eval.regular.available                       | 80 MiB
           kvflowcontrol.tokens.eval.regular.deducted                        | 0 B
           kvflowcontrol.tokens.eval.regular.returned                        | 0 B
           kvflowcontrol.tokens.eval.regular.returned.disconnect             | 0 B
        @@ -127,11 +123,11 @@
           kvflowcontrol.tokens.send.elastic.available                       | 40 MiB
           kvflowcontrol.tokens.send.elastic.deducted                        | 11 MiB
           kvflowcontrol.tokens.send.elastic.deducted.force_flush_send_queue | 0 B
           kvflowcontrol.tokens.send.elastic.deducted.prevent_send_queue     | 0 B
           kvflowcontrol.tokens.send.elastic.returned                        | 11 MiB
        -  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 2.0 MiB
        +  kvflowcontrol.tokens.send.elastic.returned.disconnect             | 7.0 MiB
           kvflowcontrol.tokens.send.elastic.unaccounted                     | 0 B
           kvflowcontrol.tokens.send.regular.available                       | 80 MiB
           kvflowcontrol.tokens.send.regular.deducted                        | 0 B
           kvflowcontrol.tokens.send.regular.deducted.prevent_send_queue     | 0 B
           kvflowcontrol.tokens.send.regular.returned                        | 0 B
        --- FAIL: TestFlowControlRaftMembershipV2/v2_enabled_when_leader_level=2/kvadmission.flow_control.mode=apply_to_elastic (60.68s)
```

Jira issue: CRDB-43528

Epic CRDB-37515

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -3078,6 +3078,11 @@ func TestFlowControlRaftMembershipV2(t *testing.T) {
 				ReplicationMode: base.ReplicationManual,
 				ServerArgs: base.TestServerArgs{
 					Settings: settings,
+					RaftConfig: base.RaftConfig{
+						// Suppress timeout-based elections. This test doesn't want to deal
+						// with leadership changing hands unless intentional.
+						RaftElectionTimeoutTicks: 1000000,
+					},
 					Knobs: base.TestingKnobs{
 						Store: &kvserver.StoreTestingKnobs{
 							FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{

